crypto/txt_db/txt_db.c

```diff
@@ -44,7 +43,7 @@ TXT_DB *TXT_DB_read(BIO *in, int num)
         goto err;
     if ((ret->qual = OPENSSL_malloc(sizeof(*(ret->qual)) * num)) == NULL)
         goto err;
-    for (i = 0; i < num; i++) {
+    for (i = 0; i < unum; i++) {
         ret->index[i] = NULL;
         ret->qual[i] = NULL;
     }
```

Comment:
If `num` is negative, then `unum` could be excessive, and this loop would overwrite memory and/or crash.